testsuite/MDAnalysisTests/coordinates/test_imd.py

```diff
@@ -0,0 +1,663 @@
+        self.topology = COORDINATES_TOPOLOGY
+        self.changing_dimensions = True
+        self.reader = IMDReader
+
+        self.first_frame.velocities = self.first_frame.positions / 10
+        self.first_frame.forces = self.first_frame.positions / 100
+
+        self.second_frame.velocities = self.second_frame.positions / 10
+        self.second_frame.forces = self.second_frame.positions / 100
+
+        self.last_frame.velocities = self.last_frame.positions / 10
+        self.last_frame.forces = self.last_frame.positions / 100
+
+        self.jump_to_frame.velocities = self.jump_to_frame.positions / 10
+        self.jump_to_frame.forces = self.jump_to_frame.positions / 100
+
+    def iter_ts(self, i):
+        ts = self.first_frame.copy()
+        ts.positions = 2**i * self.first_frame.positions
+        ts.velocities = ts.positions / 10
+        ts.forces = ts.positions / 100
+        ts.time = i
+        ts.frame = i
+        return ts
+
+
+@pytest.mark.skipif(not HAS_IMDCLIENT, reason="IMDClient not isntalled")
+class TestIMDReaderBaseAPI(MultiframeReaderTest):
+
+    @pytest.fixture()
+    def ref(self):
+        """Not a static method like in base class- need new server for each test"""
+        return IMDReference()
+
+    @pytest.fixture()
+    def reader(self, ref):
+        # This will start the test IMD Server, waiting for a connection
+        # to then send handshake & first frame
+        ref.server.handshake_sequence("localhost", ref.port)
+        # This will connect to the test IMD Server and read the first frame
+        reader = ref.reader(ref.trajectory, n_atoms=ref.n_atoms)
+        # Send the rest of the frames- small enough to all fit in socket itself
+        ref.server.send_frames(1, 5)
+
+        reader.add_auxiliary(
+            "lowf",
+            ref.aux_lowf,
+            dt=ref.aux_lowf_dt,
+            initial_time=0,
+            time_selector=None,
+        )
+        reader.add_auxiliary(
+            "highf",
+            ref.aux_highf,
+            dt=ref.aux_highf_dt,
+            initial_time=0,
+            time_selector=None,
+        )
+        return reader
+
+    @pytest.fixture()
+    def transformed(self, ref):
+        # This will start the test IMD Server, waiting for a connection
+        # to then send handshake & first frame
+        ref.server.handshake_sequence("localhost", ref.port)
+        # This will connect to the test IMD Server and read the first frame
+        transformed = ref.reader(ref.trajectory, n_atoms=ref.n_atoms)
+        # Send the rest of the frames- small enough to all fit in socket itself
+        ref.server.send_frames(1, 5)
+        transformed.add_transformations(
+            translate([1, 1, 1]), translate([0, 0, 0.33])
+        )
+        return transformed
+
+    @pytest.mark.skip(
+        reason="Stream-based reader cannot determine n_frames until EOF"
+    )
+    def test_n_frames(self, reader, ref):
```

Comment:
For all these that have skips, could you reduce the code to just a `pass`?

Reply:
Agree, skip normally denotes an issue outside of repo, e.g dependency issue.